ports/stm32/modmachine.c

```diff
@@ -155,6 +155,8 @@ STATIC mp_obj_t machine_info(size_t n_args, const mp_obj_t *args) {
         printf("ID=%02x%02x%02x%02x:%02x%02x%02x%02x:%02x%02x%02x%02x\n", id[0], id[1], id[2], id[3], id[4], id[5], id[6], id[7], id[8], id[9], id[10], id[11]);
     }
 
+    printf("REVID=%04lx\n", HAL_GetREVID());
```

Comment:
There's also HAL_GetDEVID.  And I think these two just get different parts of `DBGMCU->IDCODE`.  So maybe just print that entire register as a hex number?

Reply:
How about `printf("DEVID:%04lx REVID=%04lx\n", HAL_GetDEVID(), HAL_GetREVID());` This way they are separate ?

Reply:
@dpgeorge Might be useful to add this as well `HAL_GetHalVersion()` here or somewhere else, maybe in the REPL prompt message with the tag/build etc...

Reply:
What's the reason/use-case for adding these?  If they are useful then maybe it's better to make them easily available to code, eg as part of `os.uname()`.

Reply:
I only care about REVID because of the different H7 variants that support different frequencies. It's not easy to know which rev you have that's why I added REV-ID to info() in our fork a while ago, maybe it's more useful for developers or debugging. And it makes more sense to me in machine.info(), but feel free to change that or close the PR if you don't want it.

Reply:
I agree the REVID can be useful.  Probably `machine.freq()` needs to be aware of it for H7's so it can limit the max frequency there.  And Python code could already do `machine.mem32[0x5C001000] >> 16` to dynamically determine the revision in scripts (maybe the DBGMCU address constant can be added to `stm` to help with this...).

Anyway, DEV/REV ID seems sensible to add to this function.

Reply:
@dpgeorge Yes that's actually very important, we found some REV Ys are unstable if over-clocked to 480MHz.

Reply:
@dpgeorge Also I've been planning on sending a PR to fix something in powerctrl.c related to REV-V and voltage scaling.